libs/SalesforceSwiftSDK/SalesforceSwiftSDK/Classes/SalesforceCodables.swift

```diff
@@ -0,0 +1,111 @@
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import Foundation
+
+public struct Attributes : Codable {
+    let type : String?
+    let url : String?
+    
+    enum CodingKeys: String, CodingKey {
+        case type = "type"
+        case url = "url"
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        type = try values.decodeIfPresent(String.self, forKey: .type)
+        url = try values.decodeIfPresent(String.self, forKey: .url)
+    }
+}
+
+public class QueryResponseRecord : Codable {
+    let attributes : Attributes?
+    
+    enum CodingKeys: String, CodingKey {
+        case attributes
+    }
+    
+    required public init(from decoder: Decoder) throws {
+        let values =  try decoder.container(keyedBy: CodingKeys.self)
+        attributes = try values.decodeIfPresent(Attributes.self, forKey: .attributes)
+    }
+}
+
+public struct QueryResponse<T:QueryResponseRecord> : Codable {
+    let totalSize : Int?
+    let done : Bool?
+    let records : [T]?
+    
+    enum CodingKeys: String, CodingKey {
+        case totalSize = "totalSize"
+        case done = "done"
+        case records = "records"
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        totalSize = try values.decodeIfPresent(Int.self, forKey: .totalSize)
+        done = try values.decodeIfPresent(Bool.self, forKey: .done)
+        records = try values.decodeIfPresent([T].self, forKey: .records)
+        
+        // Decode records
+        if var allRecords = try? values.nestedUnkeyedContainer(forKey: .records) {
+            while (!allRecords.isAtEnd) {
+                let record = try allRecords.decode(T.self)
+                records?.append(record)
+            }
+        }
+    }
+}
+
+public class SearchRecord : Codable {
+    let attributes : Attributes?
+    let id : String?
+    
+    enum CodingKeys: String, CodingKey {
+        case attributes
+        case id = "Id"
+    }
+    
+    required public init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: CodingKeys.self)
+        attributes = try Attributes(from: decoder)
+        id = try values.decodeIfPresent(String.self, forKey: .id)
+    }
+    
+}
+
+struct SearchResponse : Codable {
+    let searchRecords : [SearchRecord]?
+    
+    enum CodingKeys: String, CodingKey {
+        case searchRecords = "searchRecords"
```

Comment:
Same here.

Reply:
@bhariharan There are a few new Api(s) which will get added whose output will be the above codables (SeacrhResult & QueryResult). These are all part of individual return types for the convenience apis that I will send out next.

Reply:
Right - but these codables are tied to specific API response shapes. I'd prefer we remove this (or anything that has API-response specific fields) in `6.0` or we might get stuck with supporting it for a year. Let's evaluate and reconsider in `6.1`. cc: @wmathurin